interfaces/cython/cantera/reaction.pyx

```diff
@@ -223,63 +299,52 @@ cdef class ChebyshevRate(_ReactionRate):
             for j,value in enumerate(row):
                 CxxArray2D_set(data, i, j, value)
 
-        self._base.reset(new CxxChebyshevRate3(Tmin, Tmax, Pmin, Pmax, data))
-        self.base = self._base.get()
-        self.rate = <CxxChebyshevRate3*>(self.base)
+        self._rate.reset(new CxxChebyshevRate3(Tmin, Tmax, Pmin, Pmax, data))
+        self.rate = self._rate.get()
 
-    @staticmethod
-    cdef wrap(shared_ptr[CxxReactionRateBase] rate):
-        """
-        Wrap a C++ ReactionRateBase object with a Python object.
-        """
-        # wrap C++ reaction
-        cdef ChebyshevRate arr
-        arr = ChebyshevRate(init=False)
-        arr._base = rate
-        arr.base = arr._base.get()
-        arr.rate = <CxxChebyshevRate3*>(arr.base)
-        return arr
+    cdef CxxChebyshevRate3* chr(self):
+        return <CxxChebyshevRate3*>self.rate
 
     property Tmin:
         """ Minimum temperature [K] for the Chebyshev fit """
         def __get__(self):
-            return self.rate.Tmin()
+            return self.chr().Tmin()
 
     property Tmax:
         """ Maximum temperature [K] for the Chebyshev fit """
         def __get__(self):
-            return self.rate.Tmax()
+            return self.chr().Tmax()
 
     property Pmin:
         """ Minimum pressure [Pa] for the Chebyshev fit """
         def __get__(self):
-            return self.rate.Pmin()
+            return self.chr().Pmin()
 
     property Pmax:
         """ Maximum pressure [Pa] for the Chebyshev fit """
         def __get__(self):
-            return self.rate.Pmax()
+            return self.chr().Pmax()
 
     property n_pressure:
         """ Number of pressures over which the Chebyshev fit is computed """
         def __get__(self):
-            return self.rate.nPressure()
+            return self.chr().nPressure()
 
     property n_temperature:
         """ Number of temperatures over which the Chebyshev fit is computed """
         def __get__(self):
-            return self.rate.nTemperature()
+            return self.chr().nTemperature()
 
     property coeffs:
         """
         2D array of Chebyshev coefficients of size `(n_temperature, n_pressure)`.
         """
         def __get__(self):
-            c = np.fromiter(self.rate.coeffs(), np.double)
-            return c.reshape((self.rate.nTemperature(), self.rate.nPressure()))
+            c = np.fromiter(self.chr().coeffs(), np.double)
+            return c.reshape((self.chr().nTemperature(), self.chr().nPressure()))
```

Comment:
Since those `cdef` functions are not accessible from the outside, keeping things short is imho important. To make things more consistent, I changed the various implementations to `cxx`.

Reply:
> Since those `cdef` functions are not accessible from the outside, keeping things short is imho important.

To me, it is more important to be descriptive. This code will be read many more times by future developers, so having the names be clear is important in all the code, not just in userland-API code.

Reply:
I believe that `cxx` is descriptive, as it refers to the actual C++ object.

Reply:
I think something along the lines of `cxx_object` or `cxx_pointer` or `rate_pointer` or `rate_cxx_object` would be more descriptive. I know that takes more space, but especially since this is not going to be a user-facing API, it won't have to be typed out a lot, but it will be read much more frequently, and having clarity without having to go searching for what the `cxx` means would be a benefit.

One philosophical point I've come around to the longer I've been working on Cantera is to try and avoid layers of indirection and abstraction, because there's so much going on already that it's nearly impossible to keep all of it in your brain to be able to do any work on Cantera. If it does make sense to add a layer, it should be named super clearly so that you don't need to keep referring to the implementation to know what it is.

Reply:
NP. I now have named most of the helper functions `cxx_object` (for both `ReactionRate` and `Reaction`),

FWIW, I had adopted a pre-existing approach here (which I liked and found easy to follow), but I do agree that readability is important.